dask/dataframe/core.py

```diff
@@ -579,6 +579,17 @@ def _aca_agg(self, token, func, aggfunc=None, **kwargs):
         """ Wrapper for aggregations """
         raise NotImplementedError
 
+    def apply_concat_apply(self, func, aggfunc=None, func_kwargs=None,
+                           aggfunc_kwargs=None):
+        """ Apply a function to each partition, the concat, then apply again
+        """
+        token = '-{}-{}'.format('apply_concat_apply',
```

Comment:
I used underscores instead of `apply-concat-apply` so the full name would be like `dataframe-apply_concat_apply-{sometoken}` so one would differentiate easily the "apply_concat_apply" part, but it's not important.

`_aca_agg` accepts a `token` argument that is passed over to the `apply_concat_apply` function. We could use the function names in the tokens, but people can use a lambda or reuse function names, so tokenization using the function name and arguments to disambiguate is needed anyway I believe. Right?